tests/pkg/webhook/pods.go

```diff
@@ -95,14 +97,15 @@ func admitPods(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
 	pdClient := controller.NewDefaultPDControl().GetPDClient(tc)
 	tidbController := controller.NewDefaultTiDBControl()
 
-	// if the pod is deleting, allow the pod delete operation
-	if pod.DeletionTimestamp != nil {
-		glog.Infof("pod:[%s/%s] status is timestamp %s", namespace, name, pod.DeletionTimestamp)
-		reviewResponse.Allowed = true
-		return &reviewResponse
-	}
-
 	if pod.Labels[label.ComponentLabelKey] == "tidb" {
+
+		// if tidb pod is deleting, allow pod delete operation
+		if pod.DeletionTimestamp != nil {
+			glog.Infof("TIDB pod status is namespace %s name %s timestamp %s", namespace, name, pod.DeletionTimestamp)
+			reviewResponse.Allowed = true
+			return &reviewResponse
+		}
+
```

Comment:
Why did you move this logic into tidb only. this is a common logic for tidb tikv and pd.